apps/public/src/app/(general)/bijlage/Attachments.tsx

```diff
@@ -10,36 +10,52 @@ import { postMeldingByMeldingIdAttachment } from 'apps/public/src/apiClientProxy
 
 import { BackLink } from '../_components/BackLink'
 
-import { FileUpload } from './_components/FileUpload'
+import { FileInput } from './_components/FileInput'
+import { FileList } from './_components/FileList'
 import { redirectToNextPage } from './actions'
 
+const MAX_FILES = 3
+
 type Props = {
   meldingId: number
   token: string
 }
 
-export const Bijlage = ({ meldingId, token }: Props) => {
-  const [uploadedFiles, setUploadedFiles] = useState<AttachmentOutput[]>([])
+export type UploadedFiles = AttachmentOutput & { image: string }
+
+export const Attachments = ({ meldingId, token }: Props) => {
+  const [uploadedFiles, setUploadedFiles] = useState<UploadedFiles[]>([])
   const [errorMessage, setErrorMessage] = useState<string>()
 
   const handleOnChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage(undefined)
+
     if (!event.currentTarget.files) return
 
     const files = Array.from(event.currentTarget.files)
 
+    if (files.length > MAX_FILES) {
+      setErrorMessage(`Je kunt maximaal ${MAX_FILES} bestanden uploaden.`)
+      return
+    }
+
     try {
       const result = await Promise.all(
-        files.map((file) =>
-          postMeldingByMeldingIdAttachment({
+        files.map(async (file) => {
+          const uploadedFile = await postMeldingByMeldingIdAttachment({
             formData: { file },
             meldingId,
             token,
-          }),
-        ),
+          })
+
+          const image = URL.createObjectURL(file)
```

Comment:
We should probably also [revokeObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) on deletion of a file

Reply:
this isn't necessary when using the DS FileInput